capsules/src/i2c_master_slave_driver.rs

```diff
@@ -130,7 +130,7 @@ impl hil::i2c::I2CHwMasterClient for I2CMasterSlaveDriver<'_> {
                         // than the buffer could lead to array overrun errors in
                         // userspace. The I2C syscall API should pass back lengths.
                         // -pal 3/5/21
-                        app.master_tx_buffer.mut_map_or(0, move |app_buffer| {
+                        app.master_rx_buffer.mut_map_or(0, move |app_buffer| {
```

Comment:
Was this a bug that it used `master_tx_buffer` before? Or is the idea that `WriteRead` ops should send from the tx buffer but read into the rx buffer?

Should this be accompanied by a TRD update or at least a libtock-c update to reflect which buffer is used here?

Reply:
The libtock-c documentation and interface isn't clear, but this seems like a bug.

The `MasterAction::Read` will copy the read data to the `app.master_rx_buffer` buffer while the read action from `MasterAction::WriteRead` copies data to `app.master_tx_buffer`. This changes all read operations to store data to `app.master_rx_buffer`. I'll add a comment to the libtock-c interface to make this clear

Reply:
That sounds right to me, thanks